src/Attachment.cpp

```diff
@@ -0,0 +1,52 @@
+#include <gazebo_usv/Attachment.hpp>
+
+#include <string>
+
+#include <gazebo/physics/Link.hh>
+#include <gazebo/physics/Model.hh>
+#include <gazebo/physics/World.hh>
+
+#include <gazebo_usv/Actuators.hpp>
+
+using namespace gazebo_usv;
+
+Attachment::~Attachment() {
+    delete mActuator;
+}
+
+void Attachment::load(gazebo::physics::ModelPtr model, gazebo::physics::LinkPtr link, gazebo::transport::NodePtr node) {
+    mActuator = new Actuators(model, node);
+    mLinkId = mActuator->addLink(link);
+
+    // Initialize communication node and subscribe to gazebo topic
+    std::string topicName = model->GetName() + "/" + link->GetName() + "/directional_force";
+    if (mCommandSubscriber) {
+        mCommandSubscriber->Unsubscribe();
+    }
+    mCommandSubscriber =
+        node->Subscribe("~/" + topicName, &Attachment::processDirectionalForceCommand, this);
+
+    mWorldUpdateEvent = gazebo::event::Events::ConnectWorldUpdateBegin(
+        boost::bind(&Attachment::updateBegin, this, _1)
+    );
+
+    auto worldName = GzGet((*model->GetWorld()), Name, ());
+    gzmsg << "Attachment: receiving directioned force commands from /gazebo/"
+          << worldName << "/" << link->GetName() << "/directional_force" << std::endl;
+}
+
+void Attachment::processDirectionalForceCommand(msgs::DirectionedForcePtr const& force_msg) {
+    auto direction = force_msg->direction();
+    auto force = force_msg->force();
+    
+    mForceCmd.X() = direction[0];
+    mForceCmd.Y() = direction[1];
+    mForceCmd.Z() = direction[2];
+
+    mForceCmd *= mForceCmd * force;
```

Comment:
Sorry ... I can't resist

https://www.youtube.com/watch?v=A05n32Bl0aY

Why the split ?

Reply:
> Why the split ?
It seemed easier to send the command using the two fields (magnitude and direction haha). But no problem in just reusing gazebo's vector3.proto

>And does this even compile ? Multiplying a vector by a vector ...

It does compile, but nice catch

Reply:
> It seemed easier to send the command using the two fields (magnitude and direction haha)

Then you really need to start doing more vector math ... it should really feel natural.

Reply:
> It does compile, but nice catch

Oh my ... probably cross product. I really don't like ignition...